generator/main.cpp

```diff
@@ -132,36 +137,205 @@ int main(int argc, char *argv[])
+                includes << ("/Library/Frameworks/Qt" + mod + ".framework/Headers");
+            includes << "/Library/Frameworks"; // this seems wrong
+#else
+            qWarning(
+"QTDIR environment variable not set. This may cause problems with finding the\n"
+"necessary include files.  You can find the correct directory with the\n"
+"command:\n\n"
+"  qmake -query QT_INSTALL_HEADERS\n\n"
+"This directory can be supplied using the command line argument --qt-headers\n"
+            );
+#endif
+        } else {
+            /* Legacy handling: Qt install into a single directory on its own,
+             * basically /opt/qt or something like that:
+             */
+            qtHeaders = qtdir + "/include";
+        }
+    }
+
+    if (!qtHeaders.isEmpty()) {
+        /* Look for the 'standard' directories, favouring the first found
+         * in the qt-headers list.
+         *
+         * NOTE: use of a list here is intended for the project build workflows
+         * which apparently have to assemble the headers from difference
+         * installation directories.  This could be avoided by using
+         * --include-paths instead!
+         *
+         */
+        QStringList dirList;
+        QStringList requiredModules(coreModules);
+
+        foreach (const QString &dir, qtHeaders.split(path_splitter))
+            if (QDir(dir).exists())  {
+                QStringList remaining(requiredModules);
+
+                foreach (const QString &mod, requiredModules) {
+                    const QString modpath(dir + "/Qt" + mod);
+                    if (QDir(modpath).exists()) {
+                        includes << modpath;
+                        remaining.removeAll(mod);
+                    }
+                }
+
+                requiredModules = remaining;
+                dirList << dir;
+            } else
+                qWarning((dir + " Qt header directory not found")
+                                            .toLocal8Bit().constData());
+
+        /* Add the top level directories at the end; this simulates the legacy
+         * behaviour.
+         */
+        if (!minimal)
+            includes += dirList;
+
+        /* Warn if standard directories are not found: */
+        if (!requiredModules.isEmpty()) {
+            QString sep("");
+            QString errMsg("WARNING: missing core Qt modules:");
+
+            foreach (const QString &mod, requiredModules) {
+                errMsg += sep + " Qt" + mod;
+                sep = ",";
+            }
+
+            if (args.contains("core-error"))
+                qFatal(errMsg.toLocal8Bit().constData());
+            else
+                qWarning(errMsg.toLocal8Bit().constData());
+        }
+    } else if (includes.isEmpty()) {
+        /* This is never true unless --minimal is given because "." is always
+         * added to includes first.
+         */
+        qWarning("No directories to scan: use --include-paths or --qt-headers");
+        displayHelp(&*gs);
```

Comment:
This looks weird (I know, it is used elsewhere), I would use `displayHelp(gs.get())` for clarity instead. Matter of taste, I guess...
Also, while at it - probably better to call the function something like `displayHelpAndExit`.

Reply:
I always follow the conventions of the existing code when submitting patches.  Perhaps it might be better to do code cleanup as a separate task?

Reply:
Agreed - you are right, of course.